src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\DBAL\Connection;
+use Symfony\Bridge\Doctrine\ConnectionTimingWeakMap;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    public function __construct(private ConnectionTimingWeakMap $connectionTimingWeakMap, private readonly ContainerInterface $container, private readonly array $timingByConnection)
```

Comment:
I think we can do with just this:
```suggestion
    public function __construct(private \WeakMap $connectionTimingRegistry)
```

Reply:
Yes indeed

Reply:
But do I need to retrieve the various timings that were defined in the configuration, right?

Reply:
OK, after reviewing https://github.com/doctrine/DoctrineBundle/pull/1739, I now get that the easiest way to wire things together is to identify connection objects by name (they're public services already).
So indeed, we don't need a WeakMap anymore (an ArrayObject is enough), and we do need the $container.
That's two arguments, so we can remove one :)

Reply:
Updated, and ContainerInterface has been replaced with ConnectionRegistry.